nvflare/app_common/workflows/statistics_controller.py

```diff
@@ -273,17 +350,70 @@ def _prepare_inputs(self, metric_task: str, fl_ctx: FLContext) -> Shareable:
 
         for tm in target_metrics:
             if tm.name == StC.STATS_HISTOGRAM:
-                inputs[StC.STATS_MIN] = self.global_metrics[StC.STATS_MIN]
-                inputs[StC.STATS_MAX] = self.global_metrics[StC.STATS_MAX]
+                if StC.STATS_MIN in self.global_metrics:
+                    inputs[StC.STATS_MIN] = self.global_metrics[StC.STATS_MIN]
+                if StC.STATS_MAX in self.global_metrics:
+                    inputs[StC.STATS_MAX] = self.global_metrics[StC.STATS_MAX]
 
             if tm.name == StC.STATS_VAR:
-                inputs[StC.STATS_GLOBAL_COUNT] = self.global_metrics[StC.STATS_COUNT]
-                inputs[StC.STATS_GLOBAL_MEAN] = self.global_metrics[StC.STATS_MEAN]
+                if StC.STATS_COUNT in self.global_metrics:
+                    inputs[StC.STATS_GLOBAL_COUNT] = self.global_metrics[StC.STATS_COUNT]
+                if StC.STATS_MEAN in self.global_metrics:
+                    inputs[StC.STATS_GLOBAL_MEAN] = self.global_metrics[StC.STATS_MEAN]
 
         inputs[StC.METRIC_TASK_KEY] = metric_task
-        inputs[StC.STATS_TARGET_METRICS] = target_metrics
+
+        inputs[StC.STATS_TARGET_METRICS] = fobs.dumps(target_metrics)
 
         return inputs
 
+    def _wait_for_all_results(
+        self,
+        result_wait_timeout: float,
+        requested_client_size: int,
+        client_metrics: dict,
+        sleep_time: float = 1,
+        abort_signal=None,
+    ) -> bool:
```

Comment:
suggest make this method either static method or function outside of class.

Can just pass in the self.logger

Reply:
but we could do that in follow up PRs.